pipeline_dp/dp_computations.py

```diff
@@ -23,21 +23,35 @@
 
 
 @dataclass
-class MeanVarParams:
+class ScalarNoiseParams:
     """The parameters used for computing the dp sum, count, mean, variance."""
 
     eps: float
     delta: float
-    min_value: float
-    max_value: float
+    min_value: Optional[float]
+    max_value: Optional[float]
+    min_sum_per_partition: Optional[float]
+    max_sum_per_partition: Optional[float]
     max_partitions_contributed: int
     max_contributions_per_partition: int
     noise_kind: NoiseKind  # Laplace or Gaussian
 
+    def __post_init__(self):
+        assert self.min_value is None == self.max_value is None, "min_value and max_value should be or both set or both None."
+        assert self.min_sum_per_partition is None == self.max_sum_per_partition is None, "min_sum_per_partition and max_sum_per_partition should be or both set or both None."
+
     def l0_sensitivity(self):
         """"Returns the L0 sensitivity of the parameters."""
         return self.max_partitions_contributed
 
+    @property
+    def bounds_per_value_are_set(self) -> bool:
```

Comment:
What if client has two metrics in one AggregateParams, mean with min_value & max_value and sum with min_sum_per_partition and max_sum_per_partition?

Reply:
Why is that? Can we not support them both?